### ISSUE TITLE
Docs: Send Client Component data to Server Component

### ISSUE THREAD

**Issue by mrhrifat** [vercel/next.js#56373]
### What is the improvement or update you wish to see?

 I would like to see more examples of how to send client component data to server component by following example

- Below Component is a **Client Component**
- The Child Component or `{children}` will be a **Server Component**
```
<ClientComponent>
    <ServerComponent/>
</ClientComponent>
```

- I need to send `count` & `setCount` to that **Server Component**.
- I can't do `count` & `setCount` to it's parent cause that is an Server Component.

```
'use client'
 
import { useState } from 'react'
 
export default function ClientComponent({
  children,
}: {
  children: React.ReactNode
}) {
  const [count, setCount] = useState(0)
 
  return (
    <>
      <button onClick={() => setCount(count + 1)}>{count}</button>
      {children}
    </>
  )
}
```

- Above example is already exist on [NextJS Docs](https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#supported-pattern-passing-server-components-to-client-components-as-props). But how to achieve below example context?
- I tried a lot in different way & search in docs but failed.

### Is there any context that might help us understand?

- But how to achieve below example context?

```
'use client'
 
import { useState } from 'react'
 
export default function ClientComponent({
  children,
}: {
  children: React.ReactNode
}) {
  const [count, setCount] = useState(0)
 
  return (
    <>
      <button onClick={() => setCount(count + 1)}>{count}</button>
      {children count={count} setCount={setCount}}
    </>
  )
}
```

### Does the docs page already exist? Please link to it.

https://nextjs.org/docs/app/building-your-application/rendering/composition-patterns#supported-pattern-passing-server-components-to-client-components-as-props

**Comment by Dragate** [comment]
As far as I understand, I doubt that's possible. So far there are no such examples. The server component might be passed as `children` to a client component, but in the virtual DOM it remains a descendant of their parent, which is a server component. The client component simply wraps around its children and does not come inbetween the server component and their parent, if that makes any sense. Their world is entirely separate, and thus the server component and the wrapper client component cannot exchange information, at least not in the react way.

An alternative is to lift-up the state of your client component to the URL as search params, which the server component does have access to through dynamic rendering. Given that state does not contain sensitive information and you validate it, it has its merits, such as potentially more URLs exposed for search engines, direct navigation to a specific "state" of the page from the site itself or externally (e.g. bookmarks, social sharing). If no state remains in JS, you might even convert the client component to a server one, which boosts performance.

**Comment by mrhrifat** [comment]
JajakAllah Khayer... I tried different way including **HOC, Render Props, Lift in State Up** & Others. But it's make the process more nested but don't solve the issue. Always get stucked into **Server & Client Component** issue.

But I solved the issue by trying to use the `state` everywhere I need & then composed **Server & Client Component**. And another with `Search Params` although it wasn't necessary.
